src/bootstrap/bin/rustc.rs

```diff
@@ -119,17 +119,25 @@ fn main() {
         cmd.arg(format!("-Cdebuginfo={}", debuginfo_level));
     }
 
-    if env::var_os("RUSTC_DENY_WARNINGS").is_some() &&
-       env::var_os("RUSTC_EXTERNAL_TOOL").is_none() {
+    if env::var_os("RUSTC_EXTERNAL_TOOL").is_none() {
         // When extending this list, add the new lints to the RUSTFLAGS of the
         // build_bootstrap function of src/bootstrap/bootstrap.py as well as
         // some code doesn't go through this `rustc` wrapper.
-        cmd.arg("-Dwarnings");
-        cmd.arg("-Drust_2018_idioms");
-        cmd.arg("-Dunused_lifetimes");
-        if use_internal_lints(crate_name) {
-            cmd.arg("-Zunstable-options");
-            cmd.arg("-Drustc::internal");
+        if env::var_os("RUSTC_DENY_WARNINGS").is_some() {
+            cmd.arg("-Dwarnings");
+            cmd.arg("-Drust_2018_idioms");
+            cmd.arg("-Dunused_lifetimes");
```

Comment:
Only `-Dwarnings` needs to be gated on `RUSTC_DENY_WARNINGS`. If you pass `-Wunused_lifetimes -Dwarnings` then `unused_lifetimes` is still treated as deny so we can just always pass `-Wunused_lifetimes`.

I think it would be worth synchronizing the behavior here with bootstrap.py as well: https://github.com/rust-lang/rust/blob/b9de4ef89e0e53099a084001b26ec3207c5f8391/src/bootstrap/bootstrap.py#L634-L635